### DISCUSSION TITLE
Different outputs every time the model runs

### DISCUSSION THREAD

**Discussion by jonasvit** [pgmpy/pgmpy#1534] [Q&A]
Hello. I have build my Bayesian Network model which works perfectly most of the time. For scoring a specific node I use evaluation function `query` but I have noticed that sometimes it scores different results in some of the cases which doesn't make any sense to me because I do not do any changes in the code, I just literally refresh the python kernel and run the scoring process.

I have run the same code 5 times and this is the snippet of results:
![image](https://user-images.githubusercontent.com/89067278/173050843-8337d5ae-9985-4edf-b067-dfc6bb3ee07f.png)

I have also noticed that these different scorings appear for specific parameter which turns other parameters on if specific node values are on. In any case, it still does not make me any sense why it scores differently over and over again...

I would really appreciate if someone would know why this is happening.

**Comment by ankurankan** [comment]
@jonasvit Does each run of your code also learn the model structure? Because in that case model structure can be slightly different depending on the algorithm.

**Reply by ankurankan** [reply]
@jonasvit And if that's not the case, would it be possible for you to share your code so that I can reproduce this?

**Reply by jonasvit** [reply]
@ankurankan, yes, the code right now is written in a way that it would run the model building from the beginning. But why does it change model structure?
To answer your second question, can I send you my codes to your email?

**Reply by jonasvit** [reply]
@ankurankan also, when do I really know when I get the right model structure? I need to be sure that I save the structure that I wanted to implement

**Reply by jonasvit** [reply]
@ankurankan I have emailed you with my questions and the code. If you have a chance to check it, I would really appreciate it. :) Thank you!
